box-browse-sdk/src/main/java/com/box/androidsdk/browse/adapters/BoxItemAdapter.java

```diff
@@ -246,32 +241,37 @@ public void bindItem(BoxItem item) {
          * {@link BoxBrowseActivity#createBrowseFolderFragment(BoxItem, BoxSession)}
          *
          * @param holder the BoxItemHolder
+         * @param itemToBind
          */
-        protected void onBindBoxItemViewHolder(BoxItemAdapter.BoxItemViewHolder holder) {
-            if (holder.getItem() == null) {
+        protected void onBindBoxItemViewHolder(BoxItemViewHolder holder, BoxItem itemToBind) {
+            if (itemToBind == null) {
                 return;
             }
 
-            final BoxItem item = holder.getItem();
-            boolean isEnabled = mListener.getItemFilter() == null || mListener.getItemFilter().isEnabled(item);
-
-            holder.getNameView().setText(item.getName());
-            String description = "";
-            if (item != null) {
-                String modifiedAt = item.getModifiedAt() != null ?
-                        DateFormat.getDateInstance(DateFormat.MEDIUM).format(item.getModifiedAt()).toUpperCase() :
+            final BoxItem prevItem = holder.getItem();
+            boolean isSame = prevItem != null && prevItem.getId() != null &&
+                    prevItem.getId().equals(itemToBind.getId()) &&
+                    prevItem.getModifiedAt() != null &&
+                    prevItem.getModifiedAt().equals(itemToBind.getModifiedAt()) &&
+                    prevItem.getSize() != null &&
+                    prevItem.getSize().equals(itemToBind.getSize());
+            if (!isSame) {
+                holder.getNameView().setText(itemToBind.getName());
+                String modifiedAt = itemToBind.getModifiedAt() != null ?
+                        DateFormat.getDateInstance(DateFormat.MEDIUM).format(itemToBind.getModifiedAt()).toUpperCase() :
                         "";
-                String size = item.getSize() != null ?
-                        localFileSizeToDisplay(item.getSize()) :
+                String size = itemToBind.getSize() != null ?
+                        localFileSizeToDisplay(itemToBind.getSize()) :
                         "";
-                description = String.format(Locale.ENGLISH, "%s  • %s", modifiedAt, size);
-                mController.getThumbnailManager().loadThumbnail(item, holder.getThumbView());
+                String description = String.format(Locale.ENGLISH, "%s  • %s", modifiedAt, size);
+                holder.getMetaDescription().setText(description);
+                mController.getThumbnailManager().loadThumbnail(itemToBind, holder.getThumbView());
```

Comment:
Can we modify the loadThumbnail method to search to see if it already has the icon downloaded?